Antorus/Argus.lua

```diff
@@ -757,6 +761,53 @@ end
 
 function mod:GiftoftheLifebinder(args)
 	self:Message("stages", "Positive", "Long", args.spellName, args.spellId)
+	self:RegisterUnitEvent("UNIT_POWER", nil, "boss3") -- boss1 = Argus, boss2 = Khaz'goroth, boss3 = Gift of the Lifebinder
+end
+
+function mod:UNIT_POWER(unit)
+	local power = UnitPower(unit) / UnitPowerMax(unit) * 100
+	if power <= 10 then
+		self:UnregisterUnitEvent("UNIT_POWER", unit)
+		self:Message("stages", "Positive", "Long", CL.soon:format(self:SpellName(256399)), 256399) -- Withering Roots
+	end
+end
+
+do
+	local prev, lastAnnouncedStacks, stacks, scheduled = 0, 0, nil
+	function mod:WitheringRoots(args)
+		if self:GetOption(args.spellId) ~= 0 and (self:CheckOption(args.spellId, "HEALER") ~= false or self:Healer()) then -- follow the same format as WitheringRootsStacks for healers
+			prev = GetTime()
+			lastAnnouncedStacks = 1
+			self:Message(args.spellId, "Attention", "Alert", L.stacks:format(1, args.spellName))
+		else -- for others: just warn that the tree is now withering
+			self:Message("stages", "Positive", "Long", args.spellId)
+		end
+	end
+
+	-- helper function with access to a local variable to avoid
+	-- passing by value and rescheduling the timer
+	local function announce(self, spellId, spellName)
+		self:Message(spellId, "Attention", "Alert", L.stacks:format(stacks, spellName))
+		lastAnnouncedStacks = stacks
+		prev = GetTime()
+		scheduled = nil
+	end
+
+	function mod:WitheringRootsStacks(args)
+		if args.amount - lastAnnouncedStacks > 2 then -- normally it'll be 1, 4, 7...
+			local t = GetTime()
+			if t - prev > 1 then -- on Mythic players usually revive themselves simultaneously, we don't want to show multiple messages
+				prev = t
+				lastAnnouncedStacks = args.amount
+				self:Message(args.spellId, "Attention", "Alert", L.stacks:format(args.amount, args.spellName))
```

Comment:
Is there a reason you don't call `announce()` here?
If you want to be really safe you could add a `if scheduled then self:CancelTimer(scheduled) end` as well.

Reply:
Nope, there wasn't a good reason. :s